readthedocs/filetreediff/__init__.py

```diff
@@ -28,7 +32,120 @@
 MANIFEST_FILE_NAME = "manifest.json"
 
 
-def get_diff(version_a: Version, version_b: Version) -> FileTreeDiff | None:
+class FileTreeDiffFileStatus(StrEnum):
+    """Status of a file in the file tree diff."""
+
+    ADDED = auto()
+    DELETED = auto()
+    MODIFIED = auto()
+
+    @property
+    def emoji(self):
+        """Return an emoji representing the file status."""
+        if self == FileTreeDiffFileStatus.ADDED:
+            return "➕"
+        elif self == FileTreeDiffFileStatus.DELETED:
+            return "❌"
+        elif self == FileTreeDiffFileStatus.MODIFIED:
+            return "📝"
+        return ""
+
+
+class FileTreeDiffFile:
+    def __init__(
+        self,
+        filename: str,
+        current_version,
+        current_version_build: Build,
+        base_version,
+        base_version_build: Build,
+        status: FileTreeDiffFileStatus,
+        resolver: Resolver,
+    ):
+        self.filename = filename
+        self.current_version = current_version
+        self.current_version_build = current_version_build
+        self.base_version = base_version
+        self.base_version_build = base_version_build
+        self.status = status
+        self._resolver = resolver
+
+    @property
+    def url(self):
+        return self._resolver.resolve_version(
+            project=self.current_version.project,
+            version=self.current_version,
+            filename=self.filename,
+        )
+
+    @property
+    def base_url(self):
+        return self._resolver.resolve_version(
+            project=self.base_version.project,
+            version=self.base_version,
+            filename=self.filename,
+        )
+
+
+class FileTreeDiff:
+    def __init__(
+        self,
+        current_version: Version,
+        base_version: Version,
+        current_version_build: Build,
+        base_version_build: Build,
+        files: list[FileTreeDiffFile],
+        outdated: bool,
+    ):
+        self.current_version = current_version
+        self.current_version_build = current_version_build
+        self.base_version = base_version
+        self.base_version_build = base_version_build
+        self.files = files
+        self.outdated = outdated
+        self._resolver = Resolver()
+
+    @cached_property
+    def added(self):
+        """List of added files."""
+        return [file for file in self.files if file.status == FileTreeDiffFileStatus.ADDED]
+
+    @cached_property
+    def deleted(self):
+        """List of deleted files."""
+        return [file for file in self.files if file.status == FileTreeDiffFileStatus.DELETED]
+
+    @cached_property
+    def modified(self):
+        """List of modified files."""
+        return [file for file in self.files if file.status == FileTreeDiffFileStatus.MODIFIED]
+
+    def top_files(self, limit: int = 5) -> list[FileTreeDiffFile]:
+        """Get the top files from all added, deleted, and modified files."""
```

Comment:
How that _top_ is calculated? We should add that to the docstring.

Reply:
By the way, I don't see this method returning anything. It seems it's not finished yet.